src/transformers/models/mt5/modeling_mt5.py

```diff
@@ -14,8 +14,26 @@
 # limitations under the License.
 """ PyTorch mT5 model."""
 
-from ...utils import logging
-from ..t5.modeling_t5 import T5EncoderModel, T5ForConditionalGeneration, T5Model
+import copy
+import warnings
+from typing import Optional, Tuple, Union
+
+import torch
+from torch import nn
+from torch.nn import CrossEntropyLoss
+
+from ...modeling_outputs import BaseModelOutput, Seq2SeqLMOutput, Seq2SeqModelOutput
+from ...utils import add_start_docstrings, add_start_docstrings_to_model_forward, logging, replace_return_docstrings
+from ...utils.model_parallel_utils import assert_device_map, get_device_map
+from ..t5.configuration_t5 import T5Config
+from ..t5.modeling_t5 import (
+    DEPARALLELIZE_DOCSTRING,
+    PARALLELIZE_DOCSTRING,
+    T5_ENCODER_INPUTS_DOCSTRING,
+    T5_INPUTS_DOCSTRING,
+    T5PreTrainedModel,
+    T5Stack,
+)
```

Comment:
The goal is to not have those imports but copy those objects here with appropriate `# Copied from` statements.